### PR TITLE
Game class vote handling

### PR DESCRIPTION
**Pull request by jasulcaf**
This PR handles adding the code as well as Unit tests for the following functions:
`countVoteProcess`

### CODE REVIEW

**Comment by jasulcaf** [comment]
Extra note: Please note that PR #56 should be merged _first_! This is because this PR fixes a ton of Syntax errors and miscall functions (either in parameters, or that don't exist). Do not merge until pull next changes and ensure merge doesn't conflict.

**Comment by jasulcaf** [comment]
Another note: I currently expect the tests to fail, this is because the function I implement here _calls_ a function called in PR #56.

File: appGeo/src/app/game/game.component.ts
Comment (noahklowden) [inline appGeo/src/app/game/game.component.ts:10]: I'm confused by this numbering system for these. Given these are a boolean pair of constants, I don't understand why they're being switched to 7 and 8.

Also, we do have several constants like these we use in multiple files—we should probably move these all to a base file to import from.

File: appGeo/src/tests/game.spec.ts
Comment (noahklowden) [inline appGeo/src/tests/game.spec.ts:1]: I think all general changes to these tests will be overlapping with #55, which I think has a bit more deeper of an update on these. So we should discuss the connection of the two of these since we obviously shouldn't merge both game testing files.

File: appGeo/src/app/game/game.component.ts
Comment (jasulcaf) [reply appGeo/src/app/game/game.component.ts:10]: I did that for clarity, I'm not particularly a fan of having different constants to have the same value (could result in some tests passing when they are different things). Strictly speaking, yes, they are both boolean pair of constants. But fundamentally calling a game Inactive or Active, is different then if you are taking about Civilians or Killers.

File: appGeo/src/tests/game.spec.ts
Comment (jasulcaf) [reply appGeo/src/tests/game.spec.ts:1]: I suppose that #55 will be merged first and I will have to deal with the inevitable merge conflicts that appear. I think that what I will do is just move the tests of Game & Killing Handlers directly over to whatever comes through as a result of #55 being pushed to main. This should avoid headaches and wouldn't break the passing tests.

File: appGeo/src/tests/game.spec.ts
Comment (noahklowden) [inline appGeo/src/tests/game.spec.ts:194]: Please put a comment on why this is commented out or delete it. Otherwise we won't know why it's commented out later.

**Review by noahklowden — CHANGES_REQUESTED** [review]
Overall looks good, just add the explanation for the commented out tests so that we know what it's missing. Once that's good DM me and I'll approve.

It does seem like some of the state tests I had put in originally to ensure all states are updated properly have been cut, but I'm good to merge without those for now. If I have time I may write some more, but I'm gonna work on timer today.

**Review by noahklowden — APPROVED** [review]

**Pull request merged**
